### PR TITLE
London8-Teniola-Ogunjinrin-JavaScript3-Week1

### PR DESCRIPTION
**Pull request by teniolao**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name:Teniola
- Your City:London
- Your Slack Name:teniolao

# Homework Details

- Module:3
- Week:1

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

-----
[View rendered mandatory/thinking-like-a-programmer/task.md](https://github.com/teniolao/JavaScript-Core-3-Coursework-Week1-London8/blob/main/mandatory/thinking-like-a-programmer/task.md)

### CODE REVIEW

File: mandatory/thinking-like-a-programmer/task.md [hunk 1]
```diff
@@ -29,12 +29,21 @@ After you've watched these videos I'd like you to answer these questions
 
 ## 1. What do you think the most important quality for a programmer is?
 
-<!-- Write your answer here -->
+Problem Solving
```
Comment (Monika39-maker) [inline mandatory/thinking-like-a-programmer/task.md:32]: thats right

**Review by Monika39-maker — APPROVED** [review]

File: mandatory/debugging-practice/script.js [hunk 2]
```diff
@@ -1,9 +1,9 @@
 let myLibrary = [];
 
-window.addEventListener("load", function (e) {
-  populateStorage();
-  render();
-});
+window.addEventListener("load", (e) => {
+    populateStorage();
+    render();
+  });
```
Comment (Alirezabg) [inline mandatory/debugging-practice/script.js:6]: Hi Teniola,
Good idea!

File: mandatory/debugging-practice/script.js [hunk 3]
```diff
@@ -54,7 +54,7 @@ function render() {
   let table = document.getElementById("display");
   let rowsNumber = table.rows.length;
   //delete old table
-  for (let n = rowsNumber - 1; n > 0; n-- {
+  for (let n = rowsNumber - 1; n > 0; n--) {
```
Comment (Alirezabg) [inline mandatory/debugging-practice/script.js:57]: missing parenthesis.

File: mandatory/debugging-practice/script.js [hunk 4]
```diff
@@ -76,7 +76,7 @@ function render() {
     changeBut.className = "btn btn-success";
     cell4.appendChild(changeBut);
     let readStatus = "";
-    if (myLibrary[i].check == false) {
+    if (myLibrary[i].check) {
```
Comment (Alirezabg) [inline mandatory/debugging-practice/script.js:79]: Good idea.

File: mandatory/debugging-practice/script.js [hunk 5]
```diff
@@ -90,11 +90,11 @@ function render() {
...
-    delBut.id = i + 5;
-    cell5.appendChild(delBut);
-    delBut.className = "btn btn-warning";
-    delBut.innerHTML = "Delete";
-    delBut.addEventListener("clicks", function () {
+    delButton.id = i + 5;
+    cell5.appendChild(delButton);
+    delButton.className = "btn btn-warning";
+    delButton.innerHTML = "Delete";
+    delButton.addEventListener("click", function () {
```
Comment (Alirezabg) [inline mandatory/debugging-practice/script.js:97]: prefect

File: mandatory/thinking-like-a-programmer/task.md [hunk 6]
```diff
@@ -29,12 +29,21 @@ After you've watched these videos I'd like you to answer these questions
...
 
 ## 3. What should you do if you get stuck?
 
-<!-- Write your answer here -->
+Debug by Going through your solution step by step to find where you went wrong
+Reassess by Taking  a step back to look at the problem from another perspective
+You could delete everything and begin again with fresh eyes
+Search for solutions
+Practice by solving lots of problems
+
```
Comment (Alirezabg) [inline mandatory/thinking-like-a-programmer/task.md:49]: fantastic

**Comment by MatenRehimi** [comment]
Overall, most bugs have been spotted and fixed up. It is possible to add books without an author currently, there should be validation to check for this. Check out the validation for the other input fields to see how we may achieve this.

**Pull request closed without merging**